src/tlo/core.py

```diff
@@ -76,7 +80,7 @@ class Specifiable:
         Types.BITSET: int,
     }
 
-    def __init__(self, type_, description, categories=None):
+    def __init__(self, type_: Types, description: str, categories: Set[Any] = None):
```

Comment:
Not sure the type hint for `categories` is appropriate here - it definitely needs to at least allow for `None` types (the default) but also the docstring implies a list of strings is required rather than a set. While I think the types allowed for the `categories` argument to a pandas `Categorical` are wider than that in practice, from a brief review it looks like most (potentially) all uses in `tlo` use a list of strings so I would say `list[str] | None` would be better type hint here.